consensus/types/src/chain_spec.rs

```diff
@@ -653,19 +653,48 @@ impl ChainSpec {
         }
     }
 
-    /// Return the value of `MAX_BLOBS_PER_BLOCK` appropriate for the fork at `epoch`.
+    /// Return the value of `MAX_BLOBS_PER_BLOCK` for the given `epoch`.
+    /// NOTE: this function is *technically* not spec compliant, but
+    /// I'm told this is what the other clients are doing for `devnet-0`..
     pub fn max_blobs_per_block(&self, epoch: Epoch) -> u64 {
-        self.max_blobs_per_block_by_fork(self.fork_name_at_epoch(epoch))
+        match self.fulu_fork_epoch {
+            Some(fulu_epoch) if epoch >= fulu_epoch => {
+                let mut max_blobs_per_block = self.max_blobs_per_block_electra;
+                let mut blob_schedule = self.blob_schedule.clone();
+                blob_schedule.sort_by_key(|entry| entry.epoch);
+                for entry in blob_schedule {
+                    if epoch < entry.epoch {
+                        return max_blobs_per_block;
+                    }
+                    max_blobs_per_block = entry.max_blobs_per_block;
+                }
+                max_blobs_per_block
+            }
+            _ => match self.electra_fork_epoch {
+                Some(electra_epoch) if epoch >= electra_epoch => self.max_blobs_per_block_electra,
+                _ => self.max_blobs_per_block,
+            },
+        }
     }
 
-    /// Return the value of `MAX_BLOBS_PER_BLOCK` appropriate for `fork`.
-    pub fn max_blobs_per_block_by_fork(&self, fork_name: ForkName) -> u64 {
-        if fork_name.fulu_enabled() {
-            self.max_blobs_per_block_fulu
-        } else if fork_name.electra_enabled() {
-            self.max_blobs_per_block_electra
+    // TODO(EIP-7892): remove this once we have fork-version changes on BPO forks
+    pub fn max_blobs_per_block_within_fork(&self, fork_name: ForkName) -> u64 {
```

Comment:
I think it's worth adding some unit tests for these functions.

Reply:
ah nevermind - just read the comment above

Reply:
Added some tests. This function isn't complex right now. I plan to get rid of it. But if it stuck around for multiple forks it would be more complex.